### PR TITLE
Improves "save as Excel" functionality

### PR DESCRIPTION
**Pull request by ckaczor**
- Adds option to bold the header row
- Adds option to freeze the header row
- Adds option to enable auto filtering on the header row
- Adds option to automatically size columns based on content

Goes with sqltoolsservice PR: https://github.com/microsoft/sqltoolsservice/pull/2266

### CODE REVIEW

File: src/sql/azdata.d.ts [hunk 1]
```diff
@@ -1222,6 +1222,10 @@ declare module 'azdata' {
 		textIdentifier?: string | undefined;
 		encoding?: string | undefined;
 		formatted?: boolean | undefined;
+		freezeHeaderRow?: boolean | undefined;
```
Comment (Charles-Gagnon) [inline src/sql/azdata.d.ts:1225]: 1. Please put this in azdata.proposed.d.ts. You'll create a duplicate interface definition there with ONLY the newly added props - they will be merged into a single definition at compile time
2. Please add doc comments to these (I realize that the other ones don't have them, but we're trying to enforce that new stuff at least has them and then going back and correcting older definitions as we go)

File: src/sql/azdata.proposed.d.ts [hunk 2]
```diff
@@ -2533,4 +2533,26 @@ declare module 'azdata' {
 		 */
 		splitViewSize?: number | string | undefined;
 	}
+
+	export interface SaveResultsRequestParams {
+		/**
+		 * Freeze header row when saving as Excel.
```
Comment (Charles-Gagnon) [inline src/sql/azdata.proposed.d.ts:2539]:
```suggestion
		 * Whether to freeze the header row when saving as Excel.
```

File: src/sql/azdata.proposed.d.ts [hunk 3]
```diff
@@ -2533,4 +2533,26 @@ declare module 'azdata' {
...
+		 */
+		freezeHeaderRow?: boolean | undefined;
+
+		/**
+		 * Bold header row when saving as Excel.
```
Comment (Charles-Gagnon) [inline src/sql/azdata.proposed.d.ts:2544]:
```suggestion
		 * Whether to bold the header row when saving as Excel.
```

File: src/sql/azdata.proposed.d.ts [hunk 4]
```diff
@@ -2533,4 +2533,26 @@ declare module 'azdata' {
...
+		 */
+		boldHeaderRow?: boolean | undefined;
+
+		/**
+		 * Enable auto filter on header row when saving as Excel.
```
Comment (Charles-Gagnon) [inline src/sql/azdata.proposed.d.ts:2549]:
```suggestion
		 * Whether to enable auto filter on the header row when saving as Excel.
```

File: src/sql/azdata.proposed.d.ts [hunk 5]
```diff
@@ -2533,4 +2533,26 @@ declare module 'azdata' {
...
+		 */
+		autoFilterHeaderRow?: boolean | undefined;
+
+		/**
+		 * Auto size columns when saving as Excel.
```
Comment (Charles-Gagnon) [inline src/sql/azdata.proposed.d.ts:2554]:
```suggestion
		 * Whether to auto size columns when saving as Excel.
```

File: src/sql/workbench/contrib/query/browser/query.contribution.ts [hunk 6]
```diff
@@ -258,6 +258,26 @@ const queryEditorConfiguration: IConfigurationNode = {
 			'description': localize('queryEditor.results.saveAsExcel.includeHeaders', "When true, column headers are included when saving results as an Excel file"),
 			'default': true
 		},
+		'queryEditor.results.saveAsExcel.freezeHeaderRow': {
+			'type': 'boolean',
+			'description': localize('queryEditor.results.saveAsExcel.freezeHeaderRow', "When true, freeze header row when saving results as an Excel file"),
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/query/browser/query.contribution.ts:263]:
```suggestion
			'description': localize('queryEditor.results.saveAsExcel.freezeHeaderRow', "When true, freeze the header row when saving results as an Excel file"),
```

File: src/sql/workbench/contrib/query/browser/query.contribution.ts [hunk 7]
```diff
@@ -258,6 +258,26 @@ const queryEditorConfiguration: IConfigurationNode = {
...
+			'default': false
+		},
+		'queryEditor.results.saveAsExcel.autoFilterHeaderRow': {
+			'type': 'boolean',
+			'description': localize('queryEditor.results.saveAsExcel.autoFilterHeaderRow', "When true, enable auto filtering on header row when saving results as an Excel file"),
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/query/browser/query.contribution.ts:268]:
```suggestion
			'description': localize('queryEditor.results.saveAsExcel.autoFilterHeaderRow', "When true, enable auto filtering on the header row when saving results as an Excel file"),
```

File: src/sql/workbench/contrib/query/browser/query.contribution.ts [hunk 8]
```diff
@@ -258,6 +258,26 @@ const queryEditorConfiguration: IConfigurationNode = {
...
+			'default': false
+		},
+		'queryEditor.results.saveAsExcel.autoSizeColumns': {
+			'type': 'boolean',
+			'description': localize('queryEditor.results.saveAsExcel.autoSizeColumns', "When true, attempt to automatically size columns when saving results as an Excel file"),
+			'default': false
+		},
+		'queryEditor.results.saveAsExcel.boldHeaderRow': {
+			'type': 'boolean',
+			'description': localize('queryEditor.results.saveAsExcel.boldHeaderRow', "When true, make header row bold when saving results as an Excel file"),
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/query/browser/query.contribution.ts:278]:
```suggestion
			'description': localize('queryEditor.results.saveAsExcel.boldHeaderRow', "When true, make the header row bold when saving results as an Excel file"),
```

**Review by Charles-Gagnon — COMMENTED** [review]
Minor wording updates, but otherwise looks good (but should wait until STS change to actually be merged in)

**Review by cheenamalhotra — APPROVED** [review]
LGTM!

**Pull request merged**